src/Rule/Validate/Ip.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ *
+ * This file is part of the Aura project for PHP.
+ *
+ * @package Aura.Filter
+ *
+ * @license http://opensource.org/licenses/bsd-license.php BSD
+ *
+ */
+namespace Aura\Filter\Rule\Validate;
+
+/**
+ *
+ * Validates a value is an IP address.
+ *
+ * @package Aura.Filter
+ *
+ * @license http://opensource.org/licenses/bsd-license.php BSD
+ *
+ */
+class Ip
+{
+    /**
+     *
+     * Validates that the value is a legal IP address.
+     *
+     * @return bool True if valid, false if not.
+     *
+     */
+    public function __invoke($subject, $field)
+    {
+        $value = $subject->$field;
+
+        // This validates without regard to reserved or private ranges in both v4 and v6
+        if (filter_var($value, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4 | FILTER_FLAG_IPV6) === false) {
+            return false;
```

Comment:
filter_var is an odd duck - like strcmp 0 might be a perfectly legal return and PHP does... odd things with strings that start with 0
it's generally better to do an exact === check, since that is actual failure for the filter, not just a value PHP would consider "empty " as the return